platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/maps/renderer/glsurfaceview/MapLibreGLSurfaceView.java

```diff
@@ -1023,6 +1033,31 @@ public void queueEvent(@NonNull Runnable r) {
       }
     }
 
+    /**
+     * Wait for the queue to become empty
+     * @param timeoutMillis Timeout in milliseconds
+     * @return Number of queue items remaining
+    */
+    public int waitForEmpty(long timeoutMillis) {
```

Comment:
I would not expect the timeout to be reset, that could lead to any length of wait when the caller would reasonably expect a best-effort to return within approximately the timeout period.

If we are going to handle things being inserted from multiple threads, I think we need to add a way for the caller to make the scheduler reject new tasks, since the point is to know it's empty so you it can be destroyed, and there would always be a window between unlocking and returning where new stuff could sneak in. 

My impression is that tasks are added on the same thread that creates/destroys the pool, but there's nothing to enforce that.

Reply:
Asserting that `schedule` is called on the thread that created the pool or on one of the pooled threads seems to mostly work, but fails from `SnapshotterRenderer` in tests, which apparently runs in a separate thread.

So yeah, I guess we need a `Scheduler::close` or something.

Reply:
On reflection, we can't use a `close` in `~Context` because the background scheduler is potentially being used by other things at the same time.

But I think it's valid to say that when a `Context` is being destroyed, no new loading should be initiated, so it's enough to wait for all the already-added tasks to complete.  Any other map, etc., using the background scheduler needs to maintain its own reference to keep the pool alive.